🌱 test(e2e): Always dump kubeflex-controller-manager logs

shivansh-gohem opened the pull request:
This PR addresses issue #3183 by adding an unconditional diagnostic step to the end-to-end test workflow.

This change will help improve the debuggability of E2E tests by ensuring the `kubeflex-controller-manager` logs are always available for review, regardless of whether the test run succeeds or fails.

**Changes made:**
- Added a new step to both the `ginkgo-test` and `test-bash` jobs in the `pr-test-e2e.yml` workflow.
- This step runs `if: always()` to guarantee execution.

Fixes #3183

MikeSpreitzer commented:
/ok-to-test

shivansh-gohem commented:
I've pushed the fix for the test failures. The workflows are now awaiting approval to run. Thank you!

MikeSpreitzer commented on .github/workflows/pr-test-e2e.yml:
```diff
@@ -316,4 +323,4 @@ jobs:
           sleep 10
           curl http://localhost:8080/metrics
-          kill %
+          kill %
```
Not loving the red ink here.

<img width="160" height="94" alt="Image" src="https://github.com/user-attachments/assets/758e8fe2-fe72-4789-bb7d-21858a88887b" />

MikeSpreitzer replied:
It's baaack.

MikeSpreitzer reviewed (approved):
This is an improvement.
/lgtm

shivansh-gohem commented:
I've pushed the final corrected version of the test workflow. The checks are now ready and awaiting approval to run. Thank you for your time and help!

MikeSpreitzer commented on .github/workflows/pr-test-e2e.yml:
```diff
@@ -85,6 +84,10 @@ jobs:
 
+      - name: Dump kubeflex-controller-manager logs
+        if: always()
+        run: kubectl logs -n kubeflex-system deploy/kubeflex-controller-manager
```
(A) I would move this down later, see how there is something of a progression of dependency in the debug logging?

(B) It is helpful to also dump the log from the previous run of the same container (`-p`) if that's available.

MikeSpreitzer commented on .github/workflows/pr-test-e2e.yml:
```diff
@@ -105,6 +105,14 @@ jobs:
+          echo "--- Dumping logs for previous kubeflex-controller-manager container ---"
+          kubectl logs -n kubeflex-system deploy/kubeflex-controller-manager -p || echo "INFO: No previous container found."
+          echo "--- Dumping logs for current kubeflex-controller-manager container ---"
+          kubectl logs -n kubeflex-system deploy/kubeflex-controller-manager || echo "ERROR: Could not dump current logs."
```
The `|| echo "ERROR: Could not dump current logs."` is not really helpful. Anyone looking at the job log will see what did and did not happen.

MikeSpreitzer commented on .github/workflows/pr-test-e2e.yml:
```diff
@@ -110,6 +109,14 @@ jobs:
+        if: always()
+        run: |
+          echo "--- Dumping logs for previous kubeflex-controller-manager container ---"
+          kubectl logs -n kubeflex-system deploy/kubeflex-controller-manager -p || true
```
Sorry, I seem to have been misinformed: https://github.com/kubestellar/kubestellar/actions/runs/18973522258/job/54330340635?pr=3354 suggests that these shells _do_ implicitly have `set -e`. So the `|| true` is needed.

MikeSpreitzer commented on .github/workflows/pr-test-e2e.yml:
```diff
@@ -110,6 +109,14 @@ jobs:
+        if: always()
+        run: |
+          echo "--- Dumping logs for previous kubeflex-controller-manager container ---"
+          kubectl logs -n kubeflex-system deploy/kubeflex-controller-manager -p
```
This is already covered. See the previous job step named "show previous logs in hosting cluster".

shivansh-gohem commented:
Hi @MikeSpreitzer,

Thank you for the detailed feedback. I have followed all of your instructions:

I have rebased the branch onto the latest main to remove the merge commit and clean up the history.

The || true is now restored, as you confirmed set -e is active.

The logging step is now moved to after the "show pods" step.

The step is simplified to only log the current container, as you noted the previous logs were redundant.

The wDds1 typo is fixed.

The final newline is added.

I have pushed the single, rebased commit. The workflows are ready for approval.
